src/fhirtypes/ElementDefinition.ts

```diff
@@ -1083,6 +1086,41 @@ export class ElementDefinition {
     }
   }
 
+  /**
+   * Fixes a Reference to this element.
+   * @see {@link fixValue}
+   * @param {FshReference} value - the Reference value to fix
+   * @throws {NoSingleTypeError} when the ElementDefinition does not have a single type
+   * @throws {ValueAlreadyFixedError} when the value is already fixed to a different value
+   * @throws {TypeNotFoundError} when the value does not match the type of the ElementDefinition
+   */
+  fixFshReference(value: FshReference): void {
+    if (!this.hasSingleType()) {
+      throw new NoSingleTypeError('Reference');
+    }
+    const type = this.type[0].code;
+    if (type === 'Reference') {
+      if (this.patternReference) {
+        const found = this.patternReference;
+        const foundFshReference = new FshReference(found.reference, found.display);
+        // Check if the new quantity matches the current
+        if (!value.equals(foundFshReference)) {
+          throw new ValueAlreadyFixedError(
+            value.toString(),
+            'Reference',
+            foundFshReference.toString()
+          );
+        }
+        // If they do match, there is nothing to do, so return
+        return;
+      }
+      this.patternReference = { reference: value.reference };
+      if (value.display) this.patternReference.display = value.display;
```

Comment:
Hmm that does make sense. But I also do think it makes sense to set the `display` for references, since they are mainly (only?) going to be used on instances, and if someone sets that `display` on an instance, they probably want it to appear.

I guess in general you probably do want displays to appear if someone sets them on an instance, and don't care as much if they are set on a profile. But the both of these cases use the same fixing functions, so we either have to do some refactoring or pick one. If we have to pick one, I lean that we fix `display`, because I think people are more likely to get confused if they are putting down this value that is then not showing up in output.

Reply:
Right -- definitely want that for instances and, in fact, someone remarked that we're not preserving code displays in instances (oops).  I think we should probably change the fixed code stuff so it also fixes display if it is provided -- but that is definitely a change, so we should talk to Mark about it.